File: backend/src/main/java/com/mealplanner/ingredient/Ingredient.java
package com.mealplanner.ingredient;

import com.mealplanner.common.enums.IngredientCategory;
import com.mealplanner.common.enums.IngredientSource;
import com.mealplanner.common.enums.UnitType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "ingredients")
@Getter
@Setter
@NoArgsConstructor
public class Ingredient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(name = "open_food_facts_id")
    private String openFoodFactsId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IngredientSource source;

    @Enumerated(EnumType.STRING)
    @Column(name = "unit_type", nullable = false)
    private UnitType unitType;

    @Column(name = "piece_weight_grams")
    private Double pieceWeightGrams;

    @Column(name = "kcal_per_100g", nullable = false)
    private double kcalPer100g;

    @Column(name = "protein_per_100g", nullable = false)
    private double proteinPer100g;

    @Column(name = "carbs_per_100g", nullable = false)
    private double carbsPer100g;

    @Column(name = "fat_per_100g", nullable = false)
    private double fatPer100g;

    @Column(name = "fiber_per_100g", nullable = false)
    private double fiberPer100g;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IngredientCategory category;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "manual_override", nullable = false)
    private boolean manualOverride = false;
}

File: frontend/src/types/ingredient.ts
import type {
  IngredientCategory,
  IngredientSource,
  UnitType,
} from "@/types/enums";

export type Ingredient = {
  id: number;
  name: string;
  openFoodFactsId: string | null;
  source: IngredientSource;
  unitType: UnitType;
  pieceWeightGrams: number | null;
  kcalPer100g: number;
  proteinPer100g: number;
  carbsPer100g: number;
  fatPer100g: number;
  fiberPer100g: number;
  category: IngredientCategory;
  createdAt: string;
  manualOverride: boolean;
};

export type CreateIngredientPayload = {
  name: string;
  category: IngredientCategory;
  unitType: UnitType;
  pieceWeightGrams?: number | null;
  kcalPer100g: number;
  proteinPer100g: number;
  carbsPer100g: number;
  fatPer100g: number;
  fiberPer100g: number;
  source?: IngredientSource;
  openFoodFactsId?: string | null;
  manualOverride?: boolean;
};

export type UpdateIngredientPayload = Partial<CreateIngredientPayload>;
